src/memory.rs

```
@@ -0,0 +1,27 @@
+const MEMORY_MAX: usize = 1 << 16;
+
+pub struct Memory {
+    mem: [u16; MEMORY_MAX],
+}
+
+impl Memory {
+    pub fn new() -> Self {
+        Self {
+            mem: [0; MEMORY_MAX],
+        }
+    }
+
+    /// Safe read memory method
+    pub fn read(&self, address: u16) -> u16 {
+        let addr: usize = address.into();
+        self.mem.get(addr).copied().unwrap_or(0)
```

I would return an error if you fail to read from memory